### PR TITLE
fix: should use domain-qualified finalizer name

### PR DESCRIPTION
**Pull request by bobz965**
# Pull Request

- [ ] Make sure you have followed [Kube-OVN Code Style](https://github.com/kubeovn/kube-ovn/blob/master/CODE_STYLE.md).

## What type of this PR

Examples of user facing changes:

- Bug fixes

image

## Which issue(s) this PR fixes

Fixes #(issue-number)

### CODE REVIEW

**Comment by zhangzujian** [comment]
We should migrate the finalizer field during initialization.

**Comment by bobz965** [comment]
> We should migrate the finalizer field during initialization.

ok，

File: pkg/util/const.go [hunk 1]
```diff
@@ -3,7 +3,8 @@ package util
 const (
 	CniTypeName = "kube-ovn"
 
-	ControllerName = "kube-ovn-controller"
+	DepreciatedFinalizerName = "kube-ovn-controller"
+	FinalizerName            = "kubeovn.io/finalizer"
```
Comment (zhangzujian) [inline pkg/util/const.go:7]:
```suggestion
	FinalizerName            = "kubeovn.io/kube-ovn-controller"
```

File: pkg/controller/init.go [hunk 2]
```diff
@@ -882,3 +884,18 @@ func (c *Controller) syncFinalizers() error {
...
+		controllerutil.AddFinalizer(newObj, util.KubeOVNControllerFinalizer)
+		patch, err := util.GenerateMergePatchPayload(cachedObj, newObj)
+		if err != nil {
+			klog.Errorf("failed to generate patch payload for %s, %v", newObj.GetName(), err)
+			return nil, err
+		}
+		return patch, nil
+	}
+	return nil, nil
+}
```
Comment (zhangzujian) [inline pkg/controller/init.go:901]: We can use `https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/client#Client` to simplify codes.

**Review by zhangzujian — APPROVED** [review]

File: pkg/controller/init.go (hunk 2 above)
Comment (bobz965) [reply pkg/controller/init.go]: i tried, but it is hard for now.

**Pull request merged**